components/base/Button.tsx

```diff
@@ -2,22 +2,33 @@ import styles from "@/styles/Button.module.css";
 import { textFont } from "@/pages/_app";
 
 type ButtonVariant = "default" | "inverted";
+type ButtonBlockVariants = "unblocked" | "blocked";
 type ButtonProps = {
   variant?: ButtonVariant;
+  blockVariant?: ButtonBlockVariants;
 };
 
 const Button = ({
   variant = "default",
+  blockVariant = "unblocked",
   children,
   ...otherProps
 }: ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>) => {
+  const buttonBlockVariants = {
+    unblocked: "button-unblocked",
+    blocked: "button-blocked",
```

Comment:
Not necessary to write the type 2 kinds of variants blocked and unblocked. 

Consider using a boolean just called "block", and making it default to false on the props parameter. If devs using your Button component, they can import this component and add a "block" to change this Button to width 100% type. If they don't use "block" the button shows normal width.

Reply:
Ok, I will modify this part.